animation_nodes/nodes/generic/mix_data.py

```diff
@@ -25,27 +43,75 @@ class MixDataNode(bpy.types.Node, AnimationNode):
         description = "Clamp factor between 0 and 1",
         default = False, update = executionCodeChanged)
 
+    useFactorList: VectorizedSocket.newProperty()
+    useAList: VectorizedSocket.newProperty()
+    useBList: VectorizedSocket.newProperty()
+
     def create(self):
-        self.newInput("Float", "Factor", "factor")
-        self.newInput(self.dataType, "A", "a")
-        self.newInput(self.dataType, "B", "b")
-        self.newOutput(self.dataType, "Result", "result")
+        self.newInput(VectorizedSocket("Float", "useFactorList", ("Factor", "factor"),
+            ("Factors", "factor")))
+        self.newInput(VectorizedSocket(self.dataType, ["useAList"], ("A", "a"), ("As", "a")))
+        self.newInput(VectorizedSocket(self.dataType, ["useBList"], ("B", "b"), ("Bs", "b")))
+
+        self.newOutput(VectorizedSocket(self.dataType, ["useAList", "useBList", "useFactorList"],
+            ("Result", "result"), ("Results", "results")))
 
     def draw(self, layout):
         layout.prop(self, "clampFactor")
 
     def drawLabel(self):
+        if any([self.useAList, self.useBList, self.useFactorList]):
+            return nodeTypes[self.outputs[0].dataType.strip("List").rstrip()]
         return nodeTypes[self.outputs[0].dataType]
 
-    def getExecutionCode(self, required):
-        if self.clampFactor:
-            yield "f = min(max(factor, 0.0), 1.0)"
+    def getExecutionFunctionName(self):
```

Comment:
Keep this as a `getExecutionCode`, the single case should be as before. As for the multiple case, yield `self.execute_MixDataList(...)`.